charters/PORTABILITY/charter.md

```diff
@@ -0,0 +1,57 @@
+# Pipeline Portability Service
+
+## Description
+
+The portability service enables the confirmation that pipelines developed for the HCA DCP can be successfully ran outside of the HCA DCP.
+
+## Objectives
+
+The portability service provides the ability to run HCA DCP pipelines 
+
+## In-scope
+
+* Programmatic authorized access to submit pipelines that will be ran to confirm they execute to completion on different computing infrastructure (execution environments)
+* Programmatic authorized access to extend the backend of the portability service to include additional execution environments
+* A UI to describe the current state of submitted pipelines and execution environments
```

Comment:
The first draft of the UI would simply be state. I would imagine the most interactive thing it would have is a login. This is however close to implementation detail and I want to be careful not to go into implementation. The kinds of questions it would answer are 1. what are your pipelines and did they pass or fail and in what environment 2. what are your backends, how many pipelines passed or failed. Also maybe access to your run logs. We would have to work with users for anything more. I could see it being helpful for a backend to see the kinds of errors pipelines they have in their system but that would be contingent on if the pipeline developers would be ok with them seeing that. I would imagine yes but seems like an opt-in kinda thing. Either way, this would be something we iterate forward on after the basic UI stated here.